src/diffpy/utils/scattering_objects/diffraction_objects.py

```diff
@@ -315,13 +317,15 @@ def q_to_tth(self):
         Returns
         -------
         two_theta : array
-            An array of :math:`2\theta` values in radians
+            The array of :math:`2\theta` values in radians
         """
         q = self.on_q[0]
         q = np.asarray(q)
         wavelength = float(self.wavelength)
         pre_factor = wavelength / (4 * np.pi)
-        return np.rad2deg(2.0 * np.arcsin(q * pre_factor))
+        # limit q * pre_factor to 1 to avoid invalid input to arcsin
```

Comment:
I am not sure about this.  If arcsin_value is greater than 1 something bad has gone wrong.  We probably want to raise an exception and let the user know.  It probably means they have specified a wrong wavelength of something.